app/Models/Extensions/AlbumBuilder.php

```diff
@@ -87,10 +91,63 @@ public function getModels($columns = ['*']): array
 
 			$maxTsSelect = $minTsSelect->clone()->select(DB::raw('MAX(taken_at)'));
 
-			$this->addSelect([
+			$selects = [
 				'min_taken_at' => $minTsSelect,
 				'max_taken_at' => $maxTsSelect,
-			]);
+			];
+
+			$showSubalbum = true;
+			$showSubalbumCount = false;
+			$showPhotoCount = false;
+			$isAdmin = false;
+			$userID = null;
+			switch (Configs::getValueAsString('album_decoration')) {
```

Comment:
This is too complex, do not do logic here.
Simply retrieve the count.

For example this count can be used later for statistic pages etc... As a result, making it conditional is counter productive!

Reply:
Do note that there is no secret in those, as the data is already available when you open the album.
Simply apply the cases in the front-end depending of the configuration selected, not in the back-end.

Reply:
I guess I was getting paranoid with queries and performance issues.